test/support/utils_sycl.h

```diff
@@ -132,28 +132,106 @@ make_new_policy(_Policy&& __policy)
 // create the queue with custom asynchronous exceptions handler
 static auto my_queue = sycl::queue(default_selector, async_handler);
 
-// Invoke op(policy,rest...) for each possible policy.
+// check fp16 and fp64 support
+template<typename T>
+bool has_type_support(const sycl::device& device)
+{
+    if(std::is_same<T, double>::value)
+    {
+        return device.has(sycl::aspect::fp64);
+    }
+    else if(std::is_same<T, sycl::half>::value)
+    {
+        return device.has(sycl::aspect::fp16);
+    }
+    return true;
+}
+
+// TODO: use c++11 fatures only (see void_t and fold expressions)
```

Comment:
TODO ?

Reply:
I will do it in this patch. This is a reminder for me and a note for others reviewing the patch.